src/components/List.js

```diff
@@ -26,12 +34,67 @@ const List = ({ token }) => {
   };
 
   const handleCheckbox = async (event) => {
-    const currentDate = Date.now(); // changed to using Date.now(), time given in milliseconds
+    const currentDate = Date.now(); // time given in milliseconds
     const queryCollection = await db.collection(token).doc(event.target.id);
+    // listItem is our current snapshot of the doc
+    const listItem = docs.filter((doc) => doc.id === queryCollection.id)[0];
+    const previouslyPurchasedDate = listItem.lastPurchased;
+    const latestInterval = currentDate - previouslyPurchasedDate;
+    // If item has not been purchased set to 1, else increment by 1
+    const numberOfPurchases = !listItem.numberOfPurchases
```

Comment:
Way to check for the `numberOfPurchases` being undefined 👍 

In many programming languages, there's a thing called a "guard" statement that allows you to express this kind of logic in a more compact way. JS has this feature through the use of our boolean operators (e.g. `||` or `&&`).

You can use those operators to check for certain values truthiness just like we're doing here, but with a shorter syntax.
That would look like:
```
    const numberOfPurchases = (listItem.numberOfPurchases || 0) + 1;
```

This is just a code style kind of thing, so definitely optional. I just wanted to make you aware of it, if you didn't already know! Here's a blog post that goes into more detail if you're curious how it works: https://seanmonstar.com/post/707078771/guard-and-default-operators

Reply:
I did not know about this. Thanks!